### PR TITLE
Fix view data assign

### PR DESCRIPTION
**Pull request by JackCaoG**
This is to fix https://github.com/pytorch/xla/issues/3392. More detail debugging in https://github.com/pytorch/xla/issues/3392#issuecomment-1058916309

FYI @[ronghanghu](https://github.com/ronghanghu)

### CODE REVIEW

File: torch_xla/csrc/tensor.cpp [hunk 1]
```diff
@@ -619,16 +619,18 @@ void XLATensor::SetXlaData(xla::ComputationClient::DataPtr xla_data,
...
-    // within an in-place execution context, and we need to update the view's
+  if (data()->view != nullptr && inplace) {
+    // If we have an active view, SetIrValue() happens, and we are
+    // within an in-place execution context, we need to update the view's
     // alias as well.
     data()->view = UpdateView(data()->view, std::move(ir_value));
     data()->generation += 1;
   } else {
+    // Reset the view if we are not within an in-place execution context
+    data()->view = nullptr;
```
Comment (bdhirsh) [inline torch_xla/csrc/tensor.cpp:633]: Technically do we need to reset `data()->generation` to zero too? Maybe that doesn't end up mattering in practice though.

File: test/test_operations.py [hunk 2]
```diff
@@ -1231,6 +1231,32 @@ def test_inplace_view_non_contig(self):
...
 
+  def test_view_data_update(self):
+    a = torch.zeros(4, device=xm.xla_device())
+    v = a.view(2, 2)
+    a.data = a.data + 1
+    self.assertEqual(a.tolist(), [1, 1, 1, 1])
+    # Upadting a.data should not update v's value.
+    self.assertEqual(v.tolist(), [[0.0, 0.0], [0.0, 0.0]])
+
+  def test_view_out_computation(self):
```
Comment (bdhirsh) [inline test/test_operations.py:1242]: hmm was the code in this test actually broken before? Or are you just adding this test for completeness. (this code doesn't use the `tensor.data` field, so I wouldn't expect it to be affected by your change to `XLATensor::ShallowCopyTo`)

Comment (JackCaoG) [reply test/test_operations.py:1242]: nope, just a test case I think is nice to have

File: torch_xla/csrc/tensor.cpp (hunk 1 above)
Comment (JackCaoG) [reply torch_xla/csrc/tensor.cpp:633]: right, I will reset that as well

**Review by bdhirsh — APPROVED** [review]
LGTM, nice catch :)

**Comment by ronghanghu** [comment]
I tried out the nightly PyTorch XLA wheel. This is working quite well!

**Pull request merged**